RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractRecycleBin.java

```diff
@@ -0,0 +1,552 @@
+        FsContent fsContent = (FsContent)rFile;
+        
+        skCase.addFileSystemFile(fsContent.getFileSystemId(), 
+                                rFile.getDataSourceObjectId(),
+                                rFile.getAttrType(),
+                                rFile.getAttributeId(),
+                                fileName,
+                                TskData.TSK_FS_NAME_FLAG_ENUM.UNALLOC,
+                                rFile.getSize(),
+                                0,0,0,deletedTime,
+                                rFile.getMetaAddr(),
+                                (int)rFile.getMetaSeq(),
+                                true, parentDir);
+    }
+    
+    /**
+     * 
+     * @param path
+     * @return 
+     */
+    String normalizeFilePath(String pathString) {
+        if (pathString == null || pathString.isEmpty()) {
+            return null;
+        }
+        
+        Path path = Paths.get(pathString);
+        Path rootPath = path.getRoot();
+        
+        String rootless = pathString.replace(rootPath.toString().replace("\\", ""), "");
+        return rootless.replace("\\", "/");
+    }
+    
+    String getFileName(String filePath) {
+        Path fileNamePath = Paths.get(filePath).getFileName();
+        if(fileNamePath != null) {
+            return fileNamePath.toString();
+        }
+        return filePath;
+    }
+    
+    String getParentPath(String path) {
+        Path parentPath = Paths.get(path).getParent();
+        if(parentPath != null) {
+            return normalizeFilePath(parentPath.toString());
+        }
+        return null;
+    }
+    
+    /**
+     * Stores the data from the $I files.
+     */
+    final class RecycledFileMetaData {
+
+        private final long fileSize;
+        private final long deletedTimeStamp;
+        private final String fileName;
+
+        /**
+         * Constructs a new instance.
+         *
+         * @param fileSize         Size of the deleted file.
+         * @param deletedTimeStamp Time the file was deleted.
+         * @param fileName         Name of the deleted file.
+         */
+        RecycledFileMetaData(Long fileSize, long deletedTimeStamp, String fileName) {
+            this.fileSize = fileSize;
+            this.deletedTimeStamp = deletedTimeStamp;
+            this.fileName = fileName;
+        }
+
+        /**
+         * Returns the size of the recycled file.
+         * 
+         * @return Size of deleted file
+         */
+        long getFileSize() {
+            return fileSize;
+        }
+
+        /**
+         * Returns the time the file was deleted.
+         * 
+         * @return deleted time in epoch seconds.
+         */
+        long getDeletedTimeStamp() {
+            return deletedTimeStamp;
+        }
+
+        /**
+         * Returns the name of the deleted file.
+         * 
+         * @return String name of the deleted file
+         */
+        String getFileName() {
```

Comment:
Is this full path (including "C:\") or just the final file name?  We should clarify and perhaps rename.

Reply:
This is just the last bit of the path.  
If
C:\Temp\Directory\Folder2
Folder2 will be returned

If 
C:\Temp\Directory\Folder2\file.txt
file.txt will be returned.

I have added a proper function header with the description, but I was staring at the function trying to decide whether it needed a better name.  The current name was mirroring the function from Path.